app/controllers/papers_controller.rb

```diff
@@ -5,7 +5,10 @@ class PapersController < ApplicationController
   # GET /papers
   # GET /papers.json
   def index
-    @papers = Paper.all
+    @papers = Paper.select('papers.*,sum(cached_votes_up) as total_votes')
+    .joins(:references)
+    .group('papers.id')
+    .order(('total_votes DESC, ' if params[:sort] != 'pub_date').to_s + 'papers.published_at DESC NULLS LAST')
```

Comment:
sure.  Select selects the columns you want.  paper.* selects all the columns in the papers table and sum(cached_votes_up) sums up all the votes based on what you group by.  I'm joining the references table to the papers tables and grouping by papers.id so it's giving me one paper per result and summing all the votes that way, which is all votes on each of the references associated with that paper.  The order sorts by the total votes for that paper and then by publication date.